### PR TITLE
Built-in symbols are "<builtin>", not "<command-line>"

### PR DESCRIPTION
**Pull request by Rangi42**
Some tinkering while trying to change how `FileStackNode`s are memory-managed.

### CODE REVIEW

File: src/asm/symbol.cpp [hunk 1]
```diff
@@ -563,25 +555,17 @@ void sym_SetExportAll(bool set) {
 	exportAll = set;
 }
 
-static Symbol *createBuiltinSymbol(std::string const &symName) {
-	Symbol *sym = &createsymbol(symName);
-
-	sym->isBuiltin = true;
-	sym->src = nullptr;
-	sym->fileLine = 1; // This is 0 for CLI-defined symbols
```
Comment (ISSOtm) [inline src/asm/symbol.cpp:571]: This hasn't been carried over, though?

File: src/asm/symbol.cpp [hunk 2]
```diff
@@ -90,40 +89,39 @@ std::string *Symbol::getEqus() const {
...
+	if (sym.src) {
 		sym.src->dump(sym.fileLine);
-	else if (sym.fileLine == 0)
-		fputs("<command-line>", stderr);
-	else
-		fputs("<builtin>", stderr);
-}
-
-// Set a symbol's definition filename and line
-static void setSymbolFilename(Symbol &sym) {
```
Comment (ISSOtm) [inline src/asm/symbol.cpp:102]: Why remove this function?

File: src/asm/symbol.cpp (hunk 1 above)
Comment (Rangi42) [reply src/asm/symbol.cpp:571]: The 0/1 distinction was only being used for printing `"<builtin>"` vs `"<command-line>"`, and it was doing so incorrectly. The `isBuiltin` flag is appropriate for this.

File: src/asm/symbol.cpp (hunk 2 above)
Comment (Rangi42) [reply src/asm/symbol.cpp:102]: It's an extra level of indirection that was making it that little bit harder to see how the fstk `referenced` and `ID` values are propagating. Also it was only a two-line function that's called three times, and one of those three calls was subsequently overwriting one of the two lines (macros set their own `fileLine`).

**Pull request merged**